### ISSUE TITLE
FinderUtil improvement

### ISSUE THREAD

**Issue by kantenkugel** [jagrosh/JDA-Utilities#31]
While going through the `FinderUtil` code to find out what exactly it does, I found a lot of code of the form
```java
if(matcher.matches()) {
  String groupX = matcher.replaceAll("$X");
  ...
```
which is essentially equivalent to just getting the captured group via `matcher.group(X)`.

I'm not exactly sure if changing it would result in much more performance tho (`replaceAll` docs mention a reset process)

[Example reference code](https://github.com/JDA-Applications/JDA-Utilities/blob/master/src/main/java/com/jagrosh/jdautilities/utils/FinderUtil.java#L269-L272)

**Comment by Shengaero** [comment]
We'll fix this for 1.7 👌

**Comment by Shengaero** [comment]
After some investigation it seems as if this issue has more merit than I originally expected:

From looking at the source code for both `Matcher#group(int)` and `Matcher#replaceAll(String)`, `group` appears to not only return the same thing except it does it via `String#substring(int, int)` whereas `replaceAll` does it via creation of a `StringBuffer` and a do-while loop.

**Comment by Shengaero** [comment]
Addressed in **[63a8395](https://github.com/JDA-Applications/JDA-Utilities/commit/63a839515af6c72158c1f8e20e57d07b99628c05)**
